src/Microsoft.Azure.WebJobs.Host/FunctionInvocationContext.cs

```diff
@@ -0,0 +1,68 @@
+﻿// FunctionContext.cs
+
+using System;
+using System.Collections.Generic;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.Azure.WebJobs.Host
+{
+    /// <summary>
+    /// The context for an executed function. This needs to be expanded on later.
+    /// </summary>
+    [CLSCompliant(false)]
+    public abstract class FunctionInvocationContext
+    { 
+        /// <summary>
+        /// Constructor to set the context
+        /// </summary>
+        /// <param name="id"></param>
+        /// <param name="fullName"></param>
+        /// <param name="arguments"></param>
+        /// <param name="logger"></param>
+        protected FunctionInvocationContext(Guid id, string fullName, IReadOnlyDictionary<string, object> arguments, ILogger logger)
+        {
+            Id = id;
+            FullName = fullName;
+            Arguments = arguments;
+            Logger = logger;
+        }
+
+        /// <summary>
+        /// Constructor to set the context with the JobHost
+        /// </summary>
+        /// <param name="id"></param>
+        /// <param name="fullName"></param>
+        /// <param name="arguments"></param>
+        /// <param name="logger"></param>
+        /// <param name="host"></param>
+        protected FunctionInvocationContext(Guid id, string fullName, IReadOnlyDictionary<string, object> arguments, ILogger logger, JobHost host)
+        {
+            Id = id;
+            FullName = fullName;
+            Arguments = arguments;
+            Logger = logger;
+            Host = host;
+        }
+
```

Comment:
We discussed also adding a property bag to this context object, to allow users to flow their own custom state through between their executing/executed filters. We should add:

```csharp
public IDictionary<string, object> Properties { get; }
```

Any properties added to the executing context are flowed through on the executed context.